ts/private/ts_project_worker.js

```diff
@@ -618,36 +632,46 @@ function emit(args, inputs) {
     _worker.enablePerformanceAndTracingIfNeeded();
     timingEnd('enablePerformanceAndTracingIfNeeded');
 
-    const failedLookups = new Map();
+    const changes = new Set(), creations = new Set();
+
     if (previousInputs) {
         timingStart(`invalidate`);
+        for (const [input, digest] of Object.entries(inputs)) {
+            if (!(input in previousInputs)) {
+                creations.add(input);
+            } else if (previousInputs[input] != digest) {
+                changes.add(input);
+            } else if (previousInputs[input] == null && digest == null) {
+                // Assume symlinks always change. bazel < 5.3 will always report symlinks without a digest.
```

Comment:
🤡

Reply:
I assume this is hard to write tests for? 😓

Reply:
bazel < 5.3 or Bazel <= 5.3?

Reply:
> I assume this is hard to write tests for? 😓

i did write an e2e for it.

Reply:
> bazel < 5.3 or bazel <= 5.3?

latter. i'll fix it

Reply:
https://github.com/aspect-build/rules_ts/pull/259/files#diff-314a713aa24f705d09b849d18346ab921244c08ee6c6e7cda6f75c8d6894f59cR178-R199 this case will fail on bazel 5.3 without the fix.